pyanaconda/ntp.py

```diff
@@ -107,25 +105,14 @@ def ntp_server_working(server_hostname, nts_enabled):
     :return: True if the given server is reachable and working, False otherwise
     :rtype: bool
     """
-    try:
-        # ntplib doesn't support NTS
-        if nts_enabled:
-            s = socket.create_connection((server_hostname, 4460), 2)
-            s.close()
-        else:
-            client = ntplib.NTPClient()
-            client.request(server_hostname)
-    except ntplib.NTPException:
-        return False
-    # address related error
-    except socket.gaierror:
-        return False
-    # socket related error
-    # (including "Network is unreachable")
-    except socket.error:
-        return False
+    timeout = str(NETWORK_CONNECTION_TIMEOUT)
+    directive = ["server", server_hostname, "iburst", "maxsamples", "1"]
+
+    if nts_enabled:
+        directive.append("nts")
 
-    return True
+    arguments = ["-Q", " ".join(directive), "-t", timeout]
```

Comment:
I would remove the `timeout` variable instead just add the ``str(NETWORK_CONNECTION_TIMEOUT)`` directly.

Reply:
Fixed.